### DISCUSSION TITLE
viosock driver inf sign failed through microsoft

### DISCUSSION THREAD

**Discussion by xuehuihui** [virtio-win/kvm-guest-drivers-windows#894] [Q&A] [answered]
error log：
Error 1303 in viosock.Inf, line 99 : Found legacy AddReg operation defining co-installers (CoInstallers32).
Failed INF validation. INF did not pass Desktop validation (InfVerif /k).

**Comment by xuehuihui** [comment]
drive compilation has the same error.
![image](https://user-images.githubusercontent.com/35682355/224676763-6603810b-91a6-453c-8689-789ad23c39df.png)

**Comment by xuehuihui** [comment]
@YanVugenfirer  the inf may not be written properly.  
refer to https://learn.microsoft.com/en-us/windows-hardware/drivers/devtest/inf-validation-errors-and-warnings
![image](https://user-images.githubusercontent.com/35682355/224677625-d5700b7b-9e51-408d-a6dc-018ccee83115.png)

**Comment by YanVugenfirer** [comment]
Hi @xuehuihui,

Are you using the latest WDK or the one from the WiKi? MS is moving to require only "universal" drivers. Universal drivers don't support coinstallers. The way the socket provider is registered should also change as a result.

Best regards,
Yan.

**Reply by xuehuihui** [reply]
ewdk version :EWDK_co_release_svc_prod3_22000_220715-1440

**Reply by xuehuihui** [reply]
Hi @YanVugenfirer ,
    Is there any way to circumvent it?

**Comment by YanVugenfirer** [comment, marked as answer]
Hi @xuehuihui,

We have 3 use cases:
1. Build. This is only a warning, so the package is built
2.  Attestation signing - for a month or so, Microsoft will fail the signing of a non-Universal driver. Universal drivers are not allowed to have co-installers (DLL with callbacks that will be called during different installation\uninsallation stages)
3. WHQL singing - according to our QE still works.

The coinstaller, as far as I see, is used to register the protocol during installation. So we need to make some other option to register everything that is needed. It is also possible still to pass WHQL certification.

Best regards,
Yan

**Reply by xuehuihui** [reply]
Thanks for your reply

**Comment by YanVugenfirer** [comment]
And we have the same problem for virtio-rng as well.
